models/demos/wormhole/squeezebert/demo/demo.py

```diff
@@ -0,0 +1,314 @@
+            model_name=tt_model_name,
+            initialize_model=lambda: hugging_face_reference_model,
+            custom_preprocessor=ttnn_functional_squeezebert.custom_preprocessor,
+            device=mesh_device,
+        )
+
+    tokenizer = transformers.SqueezeBertTokenizer.from_pretrained(model_name)
+    config = hugging_face_reference_model.config
+
+    nlp = transformers.pipeline("question-answering", model=hugging_face_reference_model, tokenizer=tokenizer)
+
+    attention_mask = True
+    token_type_ids = True
+    inputs_squadv2 = squadv2_1K_samples_input(tokenizer, sequence_size, attention_mask, token_type_ids, batch_size)
+    squad_metric = evaluate.load("squad_v2")
+
+    with torch.no_grad():
+        pred_labels = []
+        cpu_pred_labels = []
+        true_labels = []
+        i = 0
+        for batch in inputs_squadv2:
+            if i < n_iterations:
+                batch_data = batch[0]
+                curr_batch_size = batch_data["input_ids"].shape[0]
+                position_ids = positional_ids(config, batch_data.input_ids)
+
+                ttnn_squeezebert_inputs = squeezebert.preprocess_inputs(
+                    batch_data["input_ids"],
+                    batch_data["token_type_ids"],
+                    position_ids,
+                    batch_data["attention_mask"],
+                    device=mesh_device,
+                    mesh_mapper=inputs_mesh_mapper,
+                )
+
+                tt_output = squeezebert.squeezebert_for_question_answering(
+                    config,
+                    *ttnn_squeezebert_inputs,
+                    state_dict=state_dict,
+                    base_addr=f"transformer.",
+                    parameters=parameters,
+                    device=mesh_device,
+                    reader_patterns_cache={},
+                    mesh_mapper=inputs_mesh_mapper,
+                    mesh_composer=output_mesh_composer,
+                )
+                tt_output = ttnn.to_torch(tt_output, mesh_composer=output_mesh_composer).to(torch.float32)
+
+                cpu_output = hugging_face_reference_model(**batch_data)
+                references = batch[1]
+                question = batch[2]
+                context = batch[3]
+
+                cpu_predictions, tt_predictions = squadv2_answer_decode_batch(
+                    hugging_face_reference_model,
+                    tokenizer,
+                    nlp,
+                    references,
+                    cpu_output,
+                    tt_output,
+                    curr_batch_size,
+                    question,
+                    context,
+                )
+                pred_labels.extend(tt_predictions)
+                cpu_pred_labels.extend(cpu_predictions)
+                true_labels.extend(references)
+
+                del tt_output
+            i += 1
+        eval_score = squad_metric.compute(predictions=pred_labels, references=true_labels)
```

Comment:
it would be safer if the test fails if eval_score is lower than some threshold

Reply:
The evaluation scores should ideally remain consistent with the batch size and number of iterations specified in the demo. However, they may or may not vary with changes in batch size. I’ve now added an assertion to validate the expected scores.

Reply:
the check should be tighter. The purpose of this assert is to catch regressions if a bad commit goes in. For example, if I push a change to a kernel and the score drops from 98 to 97.9, this is a bug, and this test should catch it. 

So, we want to assert if there are any changes in the eval score, maybe with a small margin.